### PR TITLE
Allow exiting high score screen with controller start button

### PR DESCRIPTION
**Pull request by n-kall**
Currently there does not seem to be a way to exit the "enter high score" screen with a controller.
This makes it effectively impossible to quit the game with just a controller.

This PR should enable exiting the high score screen by pressing start on the controller.

### CODE REVIEW

**Comment by davidalanroth** [comment]
I'd be interested to know what controller model you are using? And is this
on the Mac? Thanks!

Best regards,

David

**Comment by n-kall** [comment]
I tested the change on Linux x86_64 using a controller that's recognised as an Xbox 360 controller.

**Comment by davidalanroth** [comment]
Thanks.

If anyone has a recommendation for a game controller they are using with
MightyMike that works well on a Mac running MacOS 14, please let us know.
Thanks!

Best regards,

David

File: src/Heart/Cinema.c [hunk 1]
```diff
@@ -607,6 +607,12 @@ ObjNode		*nameObj;
 		if (GetNewSDLKeyState(SDL_SCANCODE_RETURN) || GetNewSDLKeyState(SDL_SCANCODE_KP_ENTER))			// see if done
 			goto exit;
 
+		if (gSDLController)
+		{
+			if (SDL_GameControllerGetButton(gSDLController, SDL_CONTROLLER_BUTTON_START))
+				goto exit;
+		} 
+
```
Comment (jorio) [inline src/Heart/Cinema.c]: If possible, I'd prefer to use the key binding system than access the SDL controller directly.
I think ```GetNewNeedState(kNeed_UIPause)``` should work here.

Comment (n-kall) [reply src/Heart/Cinema.c]: Good point, I made the change as suggested

**Comment by jorio** [comment]
Thank you @n-kall!

@davidalanroth: A recent Xbox wireless controller will work fine with Mighty Mike 3.0.2 on macOS. 
Apple has an official support page about Xbox controllers that lists supported models: https://support.apple.com/en-us/111101
(Note: Xbox 360 controllers are too old and won't work on macOS)

**Pull request merged**